config/eks/config.go

```diff
@@ -110,4 +109,38 @@ func Configure(p *config.Provider) {
 		}
 		r.UseAsync = true
 	})
+	p.AddResourceConfigurator("aws_eks_access_policy_association", func(r *config.Resource) {
+		r.References = config.References{
+			"cluster_name": {
+				TerraformName: "aws_eks_cluster",
+				// Use the terraform id instead of the external name because the external name is set before the cluster
+				// has been created.
+				Extractor: common.PathTerraformIDExtractor,
```

Comment:
When I was testing this resource, I was surprised to see that with the default external name extractor, the reference to the eks cluster was being resolved even before the MR posted a Ready condition. I thought that the Ready check was part of all the cross-resource references, but when I went looking for the code that implemented it I couldn't find anything. 

I'm not sure if this is something special about the eks cluster resource, something that would be true of all resources with an external name config that has the name-based initializer enabled, something I'm just misremembering, or a sign of a larger problem. 

It seems like it would be especially problematic for those resources which use an external name which is not the same as the metadata.name, but which have an external name config that has the initializer enabled, because then references would resolve first to the metadata.name (in the external name field) and then once the real external name was set, switch to the right value.